modules/command-engine/core/src/main/scala/surge/core/KafkaProducerActor.scala

```diff
@@ -60,6 +61,62 @@ object KafkaProducerActor {
       signalBus)
   }
 
+  def createFromPartitionNumber(
+      actorSystem: ActorSystem,
+      metrics: Metrics,
+      businessLogic: SurgeModel[_, _, _, _],
+      kStreams: AggregateStateStoreKafkaStreams[_],
+      partitionTracker: KafkaConsumerPartitionAssignmentTracker,
+      signalBus: HealthSignalBusTrait,
+      config: Config,
+      kafkaProducerOverride: Option[KafkaProducerTrait[String, Array[Byte]]] = None): Int => KafkaProducerActor = partitionNumber => {
+
+    val assignedPartition = new TopicPartition(businessLogic.kafka.stateTopic.name, partitionNumber)
+
+    val brokers = config.getString("kafka.brokers").split(",").toVector
+    val adminClient = KafkaAdminClient(config, brokers)
+
+    val kafkaProducerProps = Props(
+      new KafkaProducerActorImpl(
+        assignedPartition = assignedPartition,
+        metrics = metrics,
+        businessLogic,
+        lagChecker = new KTableLagCheckerImpl(kStreams.applicationId, adminClient),
+        partitionTracker = partitionTracker,
+        signalBus = signalBus,
+        config = config,
+        kafkaProducerOverride = kafkaProducerOverride)).withDispatcher(dispatcherName)
+
+    new KafkaProducerActor(
+      ActorLifecycleManagerActor.manage(
```

Comment:
Can you use named args for all of these to make it more obvious what maps to what options?